custombackend/src/main/java/com/algolia/custombackend/elasticbackend/ElasticTransformer.java

```diff
@@ -0,0 +1,272 @@
+                    basicAuth = "Basic " + new String(android.util.Base64.encode("3nmp9kz7fh:gch2ewzerx".getBytes(), android.util.Base64.NO_WRAP));
+                }
+
+                // Open connection.
+                hostConnection = (HttpURLConnection) hostURL.openConnection();
+                hostConnection.setRequestProperty ("Authorization", basicAuth);
+
+                // Headers
+                hostConnection.setRequestProperty("Accept-Encoding", "gzip");
+                hostConnection.setRequestProperty("Content-Type","application/json");
+                hostConnection.setRequestMethod("POST");
+
+                DataOutputStream printout = new DataOutputStream(hostConnection.getOutputStream());
+                printout.writeBytes(searchParameters.toString());
+                printout.flush ();
+                printout.close ();
+
+                // read response
+                int code = hostConnection.getResponseCode();
+                final boolean codeIsError = code / 100 != 2;
+                stream = codeIsError ? hostConnection.getErrorStream() : hostConnection.getInputStream();
+                if (stream == null) {
+                    throw new IOException(String.format("Null stream when reading connection (status %d)", code));
+                }
+
+                final byte[] rawResponse;
+                String encoding = hostConnection.getContentEncoding();
+                if (encoding != null && encoding.equals("gzip")) {
+                    rawResponse = _toByteArray(new GZIPInputStream(stream));
+                } else {
+                    rawResponse = _toByteArray(stream);
+                }
+
+                // handle http errors
+                if (codeIsError) {
+                    if (code / 100 == 4) {
+                        throw new AlgoliaException(_getJSONObject(rawResponse).getString("message"), code);
+                    }
+                }
+
+                JSONObject results = _getJSONObject(rawResponse);
+                this.completionHandler.requestCompleted(results, null);
+                return results;
+
+            } catch (Exception e) { // fatal
+                Log.e("AsyncTask error", "doInBackground: " + e.getMessage());
+                this.completionHandler.requestCompleted(null, e);
+                return null;
+            } finally {
+                if (stream != null) {
+                    try {
+                        stream.close();
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }
+
+        /**
+         * Reads the InputStream as UTF-8
+         *
+         * @param stream the InputStream to read
+         * @return the stream's content as a String
+         * @throws IOException if the stream can't be read, decoded as UTF-8 or closed
+         */
+        private String _toCharArray(InputStream stream) throws IOException {
```

Comment:
WDYT of moving these to a helper in IS rather than have every app implement them? :)